### PR TITLE
http monitor: optional aggregation and last url option

### PR DESCRIPTION
**Pull request by xp-1000**
Hello @keitwb 

related to https://github.com/signalfx/signalfx-agent/pull/1459 and https://github.com/signalfx/signalfx-agent/issues/1458

thanks for taking care of that during my holidays. Nevertheless, I would like to change little the behavior.

- remove `original_url` and restore the original purpose to `url`.
- add option to add optional dimension for `last_url`, false by default to keep compat. this is sent only if different from original url.
- add url normalization to force consistency between client http config and deprecated url lists. this also allow to compare url and lastURL.
- log as debug a redirected url for troubleshooting purpose (even when `addLastURL: false`)

I understand the purpose of the previous PR and I think this one respects the logic and the ability to have a "dynamic" dimension which change when redirection changes even if it uses last url instead of original one but:

- I think it is a little less breaking now while `last_url` dimension is disabled by default and the existing `url` one has the same purpose than originally.
- avoiding dynamic change on dimension will help implement a "generic" detector for this. indeed, while `last_url` can change time to time this will generate false alert without explicit aggregation (on `url` to ignore `last_url`). without `original_url` or `last_url` we can simply not use aggregation and this will work in every cases (not matter the env, standard deployment or not, `disabledHostDimensions: true` ...)

Additional context: for now, recent changes are broke our http detectors https://github.com/claranet/terraform-signalfx-detectors/blob/master/network/http/ and I prefer to not have to use aggregation to keep compatibility everywhere.

thanks in advance 

here is a sample result:
image

with following conf: 

```yaml
  - type: http
    host: "splunk.com"
    #port: 80
    path: "/en_us/investor-relations/acquisitions/signalfx.html"
    addLastURL: true
  - type: http
    host: "signalfx.com"
    #port: 80
    addLastURL: true
  - type: http
    urls:
      - "https://www.google.com"
      - "https://google.com"
      - "http://monip.org/"
    addLastURL: true
```

### CODE REVIEW

File: pkg/monitors/http/http.go [hunk 1]
```diff
@@ -146,6 +158,29 @@ func (m *Monitor) Shutdown() {
 	}
 }
 
+func (m *Monitor) forgeURL(site string) (normalizedURL *url.URL, err error) {
```
Comment (keitwb) [inline pkg/monitors/http/http.go:161]: Call this `normalizeURL` -- "forge" is what you do to hot iron.

File: docs/monitors/http.md [hunk 2]
```diff
@@ -92,10 +95,9 @@ dimensions may be specific to certain metrics.
 
 | Name | Description |
 | ---  | ---         |
+| `last_url` | Last URL retrieved (after redirects) from configured one. Only sent if `noRedirects: false` and `addLastURL: true` and if URL responds with a redirect different from the original `url`. |
```
Comment (keitwb) [inline docs/monitors/http.md:98]: I would change everywhere it is called the "last" URL to be the "redirect" URL (e.g. `redirect_url`) instead.  This makes it immediately clear what it represents, at least clearer than "last" is.

**Comment by keitwb** [comment]
@xp-1000 Seems good except for the two minor comments I had on naming.  Normally we wouldn't do this kind of breaking changes but I think we can consider this monitor in beta still and I'm pretty sure you all at Claranet are the only ones using it so far.

**Comment by xp-1000** [comment]
> @xp-1000 Seems good except for the two minor comments I had on naming. Normally we wouldn't do this kind of breaking changes but I think we can consider this monitor in beta still and I'm pretty sure you all at Claranet are the only ones using it so far.

I totally understand this is a breaking change and I thank you for the exception. 
In fact the recent change from https://github.com/signalfx/signalfx-agent/pull/1459 is also considered as breaking change for us and this PR should allow to:

- respect the goal of the previous PR
- but resolve the breaking change inserted in v5.5.0

should be the best of both worlds I think :)

File: pkg/monitors/http/http.go (hunk 1 above)
Comment (xp-1000) [reply pkg/monitors/http/http.go:161]: ^^ sorry, my english ...

File: docs/monitors/http.md (hunk 2 above)
Comment (xp-1000) [reply docs/monitors/http.md]: should be good now

**Review by keitwb — APPROVED** [review]

**Pull request merged**